### DISCUSSION TITLE
For how long v2.3.0 RPC calls will be supported?

### DISCUSSION THREAD

**Discussion by EthanBlockson** [WalletConnect/walletconnect-monorepo#2902] [Developer Support] [answered]
We are using @web3modal/ethereum and @web3modal/react `v2.3.0` (with @wagmi/core `v^0.10.10` and wagmi `v^0.12.0`.

**How long will calls to `https://rpc.walletconnect.com/v1/?chainId=eip155:56&projectId=*` be supported under v2.6.0?**

> Migration to v2.6.0 is challenging because wagmi has removed ethers, which means that all projects using ethers need to rewrite their code due to a long list of breaking changes.

And if it will be deprecated, well, just want to know how much time we have to prepare to this hell...

**Comment by crypblizz8** [comment]
@0xAsimetriq

**Reply by xzilja** [reply, marked as answer]
Rpc calls are not affected by modal's api changes or wagmi ones, so they'll continue working on older ones. If there is a critical feature or bug we need to fix that will cause breaking changes to rpc, then there is a chance that it will stop working on older stacks.

**Reply by de702-web3** [reply]
@0xAsimetriq i believe the v1 rpc is no longer supported right now as i faced this error msg: ```
ContractFunctionExecutionError: An error occurred while executing: RPC Request failed.

URL: https://rpc.walletconnect.com/v1/?chainId=eip155:97&projectId=eb3b80135eeae2c08b3b7fcc62a34656
Request body: {"method":"eth_call","params":[{"data":"0x02571be35d184f7d19121793c20b53209ba09c57b5cc692c076f8b924621ba73988e8340"},"latest"]}

Details: stack underflow (0 <=> 2)
Version: viem@1.2.12

Raw Call Arguments:
  data:  0x02571be35d184f7d19121793c20b53209ba09c57b5cc692c076f8b924621ba73988e8340
 
Contract Call:
  function:  owner(bytes32 node)
  args:           (0x5d184f7d19121793c20b53209ba09c57b5cc692c076f8b924621ba73988e8340)

Docs: https://viem.sh/docs/contract/readContract.html
Details: stack underflow (0 <=> 2)
Version: viem@1.2.12
    at getContractError (getContractError.ts:55:10)
    at readContract (readContract.ts:100:11)
    at async ownerOfDomain (useContract.ts:80:22)
    at async isDomainAvailable (useContract.ts:107:19)
    at async Proxy.checkDomain (useMintDomainStore.ts:34:29)
```

**Reply by xzilja** [reply]
Looks like viem validation issue, did you check if actual rpc network request return some sort of error, or is it coming from viems readContrac?

**Reply by de702-web3** [reply]
i am using wagmi readContract, I tried to find the documentation of changing rpc url but i couldnt
